integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -352,11 +353,11 @@ public int read(String path, ByteBuffer buf, long size, long offset, FuseFileInf
   }
 
   private int readInternal(String path, ByteBuffer buf, long size, long offset, FuseFileInfo fi) {
+    final int sz = (int) size;
     int nread = 0;
-    int rd = 0;
     long fd = fi.fh.get();
     try {
-      FileInStream is = mOpenFileEntries.get(fd);
+      AlluxioFileInStream is = (AlluxioFileInStream) mOpenFileEntries.get(fd);
```

Comment:
Is it worth doing this cast later and reverting to the old behavior so it is compatible with the `FileInStream` API?

Reply:
done